automation_testing/tests/Components/Button/buttonPlayground.spec.ts

```diff
@@ -0,0 +1,202 @@
+		// Contained - disabled
+		await buttonPage.visit(buttonPage.page_path, [knobContained, knobDisabledTrue]);
+		expect(await buttonPage.getBackgroundColorFromElement(buttonPage.button)).toBe(expectedColor);
+	});
+
+	test("Validate the visibility of the Button with different Show options.", async () => {
+		// Undefined: Button should be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("Undefined")]);
+		await expect(buttonPage.button).toBeVisible();
+
+		// True: Button should be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("True")]);
+		await expect(buttonPage.button).toBeVisible();
+
+		// False: Button should not be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("False")]);
+		await expect(buttonPage.button).not.toBeVisible();
+
+		// Function that returns true: Button should be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("Function that returns true")]);
+		await expect(buttonPage.button).toBeVisible();
+
+		// Function that returns false: Button should not be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("Function that returns false")]);
+		await expect(buttonPage.button).not.toBeVisible();
+
+		// Array of true values: Button should be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("Array of true values")]);
+		await expect(buttonPage.button).toBeVisible();
+
+		// Array with one falsy value: Button should not be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("Array with one falsy value")]);
+		await expect(buttonPage.button).not.toBeVisible();
+
+		// Array of functions that return true: Button should be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("Array of functions that return true")]);
+		await expect(buttonPage.button).toBeVisible();
+
+		// Array of functions, one returns false: Button should not be visible.
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForShow("Array of functions, one returns false")]);
+		await expect(buttonPage.button).not.toBeVisible();
+	});
+
+	test("Validate Type of label JSX contains an Icon.", async () => {
+		await buttonPage.visit(buttonPage.page_path, [await buttonPage.getKnobForTypeOfLabel("JSX")]);
+		await expect(buttonPage.listIconLocator).toBeVisible();
+	});
+
+	test("Validate miconcolor all different variants.", async () => {
+		// yellow
+		await buttonPage.visit(buttonPage.page_path, [knob.knobShowIcon + true, await buttonPage.getKnobForIconColor("yellow")]);
+		await buttonPage.button.waitFor({state: "visible"});
+		expect.soft(await buttonPage.getColorFromElement(buttonPage.button.locator("svg"))).toBe(theme.newColors.simplyGold["100"]);
```

Comment:
After talking with Owen it was decided this is going to expect a regular yellow color.